yasnippet.el

```diff
@@ -3035,18 +3035,30 @@ through the field's start point"
    (not (and (yas--field-number field)
              (zerop (yas--field-number field))))))
 
-(defun yas--snippets-at-point (&optional all-snippets)
-  "Return a sorted list of snippets at point.
-
-The most recently-inserted snippets are returned first."
-  (sort
-   (delq nil (delete-dups
-              (mapcar (lambda (ov) (overlay-get ov 'yas--snippet))
-                      (if all-snippets (overlays-in (point-min) (point-max))
-                        (nconc (overlays-at (point))
-                               (overlays-at (1- (point))))))))
-   #'(lambda (s1 s2)
-       (<= (yas--snippet-id s2) (yas--snippet-id s1)))))
+(defun yas-active-snippets (&optional where)
+  "Return a sorted list of active snippets.
+The most recently-inserted snippets are returned first.
+
+By default, lists snippets around point.  If optional argument
+WHERE is a buffer position or pair of positions (BEG . END), list
+snippets around that position, or within the given range,
+respectively.  Any other non-nil value is equivalent to a range
+covering the whole buffer."
+  (let (beg end)
+    (unless where (setq where (point)))
+    (cond
+     ((consp where)
+      (setq beg (car where) end (cdr where)))
+     ((or (integerp where) (markerp where))
+      (setq beg where end (1+ where)))
+     (t (setq beg (point-min) end (point-max))))
+    (sort
+     (delete-dups
```

Comment:
Okay, I figured out why: snippets have multiple overlays, so getting snippets from all overlays will find the same snippet multiple times. Calling `yas--markers-to-points` on a snippet changes the format of its fields, so it leads to errors when the same snippet is processed multiple times.